ipe.go

```diff
@@ -75,6 +77,39 @@ func Start(filename string) {
 		websockets.NewWebsocket(inMemoryStorage),
 	)
 
+	router.Path("/app/create").Methods("POST").Handler(
+		api.NewCreateApplication(inMemoryStorage),
+	)
+
+	router.HandleFunc("/apps/all",
+		func(w http.ResponseWriter, r *http.Request) {
+			authHeader := r.Header.Get("Authorization")
+			log.Error(os.Getenv("AUTH_TOKEN"))
+			if authHeader != os.Getenv("AUTH_TOKEN") {
+				w.Header().Set("Content-Type", "application/json")
+				w.WriteHeader(http.StatusUnauthorized)
+				w.Write([]byte(`{ "Error":  "Unauthorized access" }`))
+				return
+			}
+
+			out, err := json.Marshal(inMemoryStorage.GetAllApps())
```

Comment:
Don't use the `inMemoryStorage` directly. Use an abstraction on top of `storage.Storage`.